File: Data Structures & Algorithms/majority-element/submission-0.go
func majorityElement(nums []int) int {
	response := 0
	count := 0

	for _, currentNum := range nums {
		if count == 0 {
			response = currentNum
		}

		if currentNum == response {
			count++
		} else {
			count--
		}
	}

	return response
}

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        response = []
        def dfs(idex: int, cur: List[int]):
            if idex == len(nums):
                response.append(cur.copy())
                return

            for i in range(idex, len(nums)):
                cur[i], cur[idex] = cur[idex], cur[i]
                dfs(idex+1, cur)
                cur[i], cur[idex] = cur[idex], cur[i]

        dfs(0, nums)
        return response
